src/server/test/web/readingsBarMeterFlow.js

```diff
@@ -7,307 +7,417 @@
+								{
+									name: "Electric kW",
+									unit: "Electric",
+									defaultGraphicUnit: "kW",
+									displayable: true,
+									gps: undefined,
+									note: "special meter",
+									file: "test/web/readingsData/readings_ri_15_days_75.csv",
+									deleteFile: false,
+									readingFrequency: "15 minutes",
+									id: METER_ID,
+								},
+							];
+							// Load the data into the database
+							await prepareTest(unitData, conversionData, meterData);
+							// Get the unit ID since the DB could use any value.
+							const unitId = await getUnitId("kW");
+							// Load the expected response data from the corresponding csv file
+							const expected = await parseExpectedCsv(
+								"src/server/test/web/readingsData/expected_bar_ri_15_mu_kW_gu_kW_st_-inf_et_inf_bd_1.csv"
+							);
+							// Create a request to the API for unbounded reading times and save the response
+							const res = await chai
+								.request(app)
+								.get(`/api/unitReadings/bar/meters/${METER_ID}`)
+								.query({
+									timeInterval: ETERNITY.toString(),
+									barWidthDays: 1,
+									graphicUnitId: unitId,
+								});
+							// Check that the API reading is equal to what it is expected to equal
+							expectReadingToEqualExpected(res, expected);
+						}
+					);
 
-                // Add B18 here
+					// Add B18 here
 
-                // Add B19 here
+					// Add B19 here
 
-                // Add B20 here
+					// Add B20 here
 
-                // Add B21 here
+					mocha.it(
+						"B21: 76 day bars (no values) for 15 minute reading intervals and flow units with +-inf start/end time & kW as kW",
+						async () => {
+							const unitData = [
+								{
+									// u4
+									name: "KW",
+									identifier: "",
+									unitRepresent: Unit.unitRepresentType.FLOW,
+									secInRate: 36,
+									typeOfUnit: Unit.unitType.METER,
+									suffix: "",
+									displayable: Unit.displayableType.NONE,
+									preferredDisplay: false,
+									note: "",
+								},
+								{
+									// u5
+									name: "KWst-inf unit",
+									identifier: "",
+									unitRepresent: Unit.unitRepresentType.FLOW,
+									secInRate: 3600,
+									typeOfUnit: Unit.unitType.UNIT,
+									suffix: "",
+									displayable: Unit.displayableType.ALL,
+									preferredDisplay: false,
+									note: "",
+								},
+							];
+							const conversionData = [
+								{
+									// c4
+									sourceName: "KW",
+									destinationName: "KWst-inf unit",
+									bidirectional: false,
+									slope: 1,
+									intercept: 0,
+									note: "KW → KWst-inf unit",
+								},
+							];
+							const meterData = [
+								{
+									name: "KW KWst-inf unit",
+									unit: "KW",
+									defaultGraphicUnit: "KWst-inf unit",
+									displayable: true,
+									gps: undefined,
+									note: "",
+									file: "test/web/readingsData/expected_bar_ri_15_mu_kW_gu_kWst-inf_et_inf_bd_76.csv",
```

Comment:
The expected readings file is specified as "readings_ri_15_days_75.csv" and this should be changed. This did not fail because the request returns no data but it does it in an unexpected way with this data so is not testing what was desired.